pkg/ebpf/tracer.go

```diff
@@ -644,16 +637,51 @@ func (t *Tracer) populatePortMapping(mp *bpflib.Map) ([]uint16, error) {
 	return closedPortBindings, nil
 }
 
-func (t *Tracer) determineConnectionDirection(conn *ConnectionStats) ConnectionDirection {
-	if conn.Type == UDP {
-		return NONE
+func (t *Tracer) setConnectionDirections(connections []ConnectionStats) {
+	connRAddrMap := make(map[util.Address][]*ConnectionStats, len(connections))
+
+	for _, conn := range connections{
+		if conn.IPTranslation == nil{
+			connRAddrMap[conn.Dest] = append(connRAddrMap[conn.Dest], &conn)
+		} else {
+			connRAddrMap[conn.IPTranslation.ReplDstIP] = append(connRAddrMap[conn.IPTranslation.ReplDstIP], &conn)
```

Comment:
This should be `conn.IPTranslation.ReplSrcIP` because the fields for IPTranslation are in reverse